src/components/Editor/Tabs.js

```diff
@@ -68,9 +82,25 @@ class Tabs extends PureComponent<Props, State> {
   }
 
   componentDidUpdate(prevProps) {
-    if (!(prevProps === this.props)) {
-      this.updateHiddenTabs();
+    if (prevProps === this.props) {
+      return;
+    }
+
+    // @todo verify its correctness.
+    if (
```

Comment:
this has very little to do with codemirror - is it affected by monaco or are you trying to help us tackle a separate hard problem. This is interesting to me because the old approach has some style calculations which affect the dom ...

Reply:
Agreed, removed this out from this PR.